DigitalLearningSolutions.Data/Services/CryptoService.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Linq;
+    using System.Security.Cryptography;
+
+    public interface ICryptoService
+    {
+        public bool VerifyHashedPassword(string? hashedPassword, string password);
+    }
+
+    public class CryptoService : ICryptoService
+    {
+        private const int SaltSize = 16;
+        private const int PBKDF2SubkeyLength = 32;
+        private const int PBKDF2IterationCount = 1000;
+
+        public bool VerifyHashedPassword(string? hashedPassword, string password)
+        {
+            if (string.IsNullOrWhiteSpace(hashedPassword))
+            {
+                return false;
+            }
+
+            var hashedPasswordBytes = Convert.FromBase64String(hashedPassword);
+
+            if (hashedPasswordBytes.Length != 1 + SaltSize + PBKDF2SubkeyLength || hashedPasswordBytes[0] != 0x00)
```

Comment:
Could this be pulled into a variable(s)? This will make the statement easier to read and remove the need for a comment.